src/make_knowledge_packs.jl

```diff
@@ -0,0 +1,222 @@
+
+# Arguments
+- model: Embedding model
+- embedding_size: Embedding dimensions
+"""
+function generate_embeddings(knowledge_pack_path::String; model::AbstractString = MODEL,
+        embedding_size::Int = EMBEDDING_SIZE)
+    embedder = RT.BatchEmbedder()
+    entries = readdir(knowledge_pack_path)
+    # Initialize a dictionary to group files by hostname and chunk size
+    hostname_files = Dict{String, Dict{Int, Dict{String, String}}}()
+
+    # Regular expressions to match the file patterns of chunks and sources
+    chunks_pattern = r"^(.*)-chunks-max-(\d+)-min-(\d+)\.jls$"
+    sources_pattern = r"^(.*)-sources-max-(\d+)-min-(\d+)\.jls$"
+
+    # chunks_pattern = r"^(.*)-chunks-(\d+)\.jls$"
+    # sources_pattern = r"^(.*)-sources-(\d+)\.jls$"
+
+    # Group files by hostname and chunk size
+    for file in entries
+        match_chunks = match(chunks_pattern, file)
+        match_sources = match(sources_pattern, file)
+
+        if match_chunks !== nothing
+            hostname = match_chunks.captures[1]
+            chunk_size = parse(Int, match_chunks.captures[2])
+            if !haskey(hostname_files, hostname)
+                hostname_files[hostname] = Dict{Int, Dict{String, String}}()
+            end
+            if !haskey(hostname_files[hostname], chunk_size)
+                hostname_files[hostname][chunk_size] = Dict{String, String}()
+            end
+            hostname_files[hostname][chunk_size]["chunks"] = joinpath(
+                knowledge_pack_path, file)
+        elseif match_sources !== nothing
+            hostname = match_sources.captures[1]
+            chunk_size = parse(Int, match_sources.captures[2])
+            if !haskey(hostname_files, hostname)
+                hostname_files[hostname] = Dict{Int, Dict{String, String}}()
+            end
+            if !haskey(hostname_files[hostname], chunk_size)
+                hostname_files[hostname][chunk_size] = Dict{String, String}()
+            end
+            hostname_files[hostname][chunk_size]["sources"] = joinpath(
+                knowledge_pack_path, file)
+        end
+    end
+    # Process each pair of files
+    for (hostname, chunk_files) in hostname_files
+        for (chunk_size, files) in chunk_files
+            if haskey(files, "chunks") && haskey(files, "sources")
+                chunks_file = files["chunks"]
+                sources_file = files["sources"]
+                chunks = deserialize(chunks_file)
+                sources = deserialize(sources_file)
+                cost_tracker = Threads.Atomic{Float64}(0.0)
+                full_embeddings = RT.get_embeddings(
+                    embedder, chunks; model, verbose = false, cost_tracker)
+                @info "Created embeddings for $hostname. Cost: \$$(round(cost_tracker[], digits=3))"
+                fn_output = joinpath(knowledge_pack_path, "packs",
+                    "$hostname-textembedding3large-0-Float32__v1.0.tar.gz")
```

Comment:
Btw `-0-Float32` implies that it's not truncated ("0"), it was the only way to distinguish in AIHelpMe if I used the same embeddings at full (3072) or truncated (1024) where the "full" size is different across models, hence, I set it to "0".